base_conditional_image/models/abstract_conditional_image.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import fields, models
+from odoo.tools.safe_eval import safe_eval
+
+
+class AbstractConditionalImage(models.AbstractModel):
+    _name = "abstract.conditional.image"
+    _description = "Abstract image"
+    _inherit = "image.mixin"
+
+    image_1920 = fields.Image(compute="_compute_images", store=False, readonly=True)
+    image_1024 = fields.Image(compute="_compute_images", store=False, readonly=True)
+    image_512 = fields.Image(compute="_compute_images", store=False, readonly=True)
+    image_256 = fields.Image(compute="_compute_images", store=False, readonly=True)
+    image_128 = fields.Image(compute="_compute_images", store=False, readonly=True)
+
+    @staticmethod
+    def _compute_selector_test_without_company(image_selector, record):
+        return bool(safe_eval(image_selector.selector or "True", {"object": record}))
+
+    @staticmethod
+    def _compute_selector_test_with_company(image_selector, record):
+        if (
+            image_selector.company_id == record.company_id
+            or record.company_id
+            and not image_selector.company_id
+        ):
+            return AbstractConditionalImage._compute_selector_test_without_company(
```

Comment:
if you want to override this method per model you'll have to override the whole module here. Either you pass this class as an argument or you use a class method instead of a static method.

Reply:
Fix